x-pack/platform/plugins/shared/inference/server/chat_complete/api.test.ts

```diff
@@ -207,6 +246,49 @@ describe('createChatCompleteApi', () => {
       ]);
     });
 
+    it('implicitly retries errors when configured to', async () => {
+      let count = 0;
+      inferenceAdapter.chatComplete.mockImplementation(() => {
+        count++;
+        if (count < 3) {
+          throw new Error(`Failing on attempt ${count}`);
+        }
+        return of(chunkEvent('chunk-1'), chunkEvent('chunk-2'));
+      });
```

Comment:
nit: duplicated code, could be a helper function

Reply:
Only 2 occurrences!

**"factorize at the 3rd occurrence for production code, and at the 5th occurrence for test code"**

*Some guy on the internet*